### PR TITLE
Router

### PR DESCRIPTION
**Pull request by emilyb7**
Just tidying up a bit
related to #80 and #138

### CODE REVIEW

File: app/router.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/* components */
+import GoalsList from './components/goals-list/goals-list.jsx';
+import AddGoal from './components/add-new-goal/add-new-goal.jsx';
+import ViewGoal from './components/view-goal/view-goal.jsx';
+
+export default (props) => {
+
+  switch (props.step) {
+    case steps.ADD_GOAL:
+      const newGoalId = (props.goals.map(g => g.id).sort((a, b) => b - a)[0] + 1 || 0);
```
Comment (des-des) [inline app/router.js]: Right now you are looking for a minimum / max. Use a reduce.

File: app/router.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      return <AddGoal
+        newGoal = { props.newGoal }
+        onInputGoal={ props.onInputGoal }
+        onSelectAvatar={ props.onSelectAvatar }
+        saveNewGoal= { props.saveNewGoal }
+        newGoalId= { newGoalId }
```
Comment (des-des) [inline app/router.js]: I am confused here. We already have something called `newGoal`, but we are also doing a search for the newGoal id on the goals. I think this is either bad naming or a mistake

**Comment by emilyb7** [comment]
@des-des , two things:

1. I just copied these bits of code over from another file and you approved them before 😛
2. I know the logic's a bit weird, but you remember that conversation we had about generating unique goal ids client-side? Well, when we get round to implementing that, this will all change anyway.

So could we just merge for now?

**Comment by des-des** [comment]
:(

**Pull request merged**